docs/tools/mdcode.py

```diff
@@ -27,13 +27,18 @@ def _parse_lines(self):
             m = _CODE_MARK.match(l)
             if m is not None:
                 if not in_code:
-                    lang = m.groups()[1].lower()
-                    indent = len(m.groups()[0])
-                yield (l, True, lang, indent)
-                if in_code:
+                    if _LANGS.has_key(m.groups()[1].lower()):
+                        lang = m.groups()[1].lower()
+                        indent = len(m.groups()[0])
+                        in_code = True
+                    elif m.groups()[1].lower() == 'bash':
+                        l = l.replace('bash', 'sh').lstrip()
```

Comment:
not sure if we want to do a special treatment for bash. is it for markdown or for notebook? i think markdown file recognize `bash`. and a notebook should not allow bash cell because it may run on windows